app/controllers/concerns/dsi_authenticatable.rb

```diff
@@ -4,8 +4,11 @@ module DsiAuthenticatable
   included do
     before_action :authenticate_dsi_user!
     before_action :handle_expired_session!
+    before_action :enforce_terms_and_conditions_acceptance!
```

Comment:
I don't think we should conflate authentication concerns with the terms and conditions flow. We could alternatively put T&Cs logic in the CheckRecordsController where it'll still apply to all CTR requests, and make the check only if `current_dsi_user` exists.